src/renderer.rs

```diff
@@ -761,11 +845,42 @@ fn humanize_systemtime(src_time: Option<SystemTime>) -> Option<String> {
 }
 
 /// Renders an error on the webpage
-pub fn render_error(error_description: &str, return_address: &str) -> Markup {
+pub fn render_error(
+    error_description: &str,
+    error_code: StatusCode,
+    return_address: &str,
+    sort_method: Option<SortingMethod>,
+    sort_order: Option<SortingOrder>,
+    color_scheme: ColorScheme,
+    default_color_scheme: ColorScheme,
+    has_referer: bool,
+) -> Markup {
+    let link = if has_referer {
+        return_address.to_string()
+    } else {
+        parametrized_link(
+            return_address,
+            sort_method,
+            sort_order,
+            color_scheme,
+            default_color_scheme,
+        )
+    };
+
     html! {
-        pre { (error_description) }
-        a href=(return_address) {
-            "Go back to file listing"
+        body {
+            (page_header(&error_code.to_string(), color_scheme, false, true))
+            div.error {
+                p { (error_code.to_string()) }
+                @for error in error_description.lines() {
+                    p { (error) }
+                }
+                div.error-nav {
+                    a.error-back href=(link) {
+                        "Go back to file listing"
```

Comment:
I'm not sure it makes sense to render this even in the case of a 401 Unauthorized error. After all, it means you cannot view the file listing because you didn't provide auth.

Reply:
Yep, that's my thought too. I'll remove it.